### PR TITLE
1264 side nav design tweaks and dynamic side nav

### LINKED ISSUE #1264: Side Nav design tweaks
# Action Requested

A few more design changes need to be made to the side navigation to match the designs:

- Vertical Alignment: the side navigation shouldn't go above the content on the right. when at the top of the page, the top of the side navigation and the top of the "eCR Summary" text should be at the same height. I attached a screenshot below to show this alignment
- Active Sections: as you scroll the page, it should be responsive to show where you are on a page with bolded blue text and a blue vertical line. [good example is on this page](https://designsystem.digital.gov/components/side-navigation/) for the component, the right side navigation is responsive as you scroll the page (... I see the styling of that component is different though and matches my original styling... perhaps there is a different component with this functionality built in??)

if the styling can actually match the USWDS component shown in this screenshot, that would be great:

let @sarahtress know if you'd like to pair at all or clarify anything with this ticket

# Acceptance Criteria

Side Navigation matches Figma design

### PR DESCRIPTION
**Pull request by nickbristow**
# PULL REQUEST

## Summary
this does the requested changes design wise to the side nav. It also makes the side nav dynamic

## Related Issue
Fixes #1264

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Comment by nickbristow** [comment]
@sarahtress for design review, 
image

**Comment by sarahtress** [comment]
@nickbristow thanks for sending the screenshot - it looks good from what I can see, but would you be able to send a video or be able to hop on a call quick (whichever is easier) to show what it looks like while you scroll? just hard to see how it's actively reflecting which section you're on in the static screenshot. thanks!

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx [hunk 1]
```diff
@@ -22,33 +22,134 @@ export class SectionConfig {
...
+          priority: headingLevels.findIndex(
+            (level) => heading.tagName.toLowerCase() == level,
+          ),
+        };
+      },
+    );
+    let sortedHeadings: SectionConfig[] = sortHeadings(headings);
+    setSectionConfigs(sortedHeadings);
+
+    let options = {
```
Comment (emmastephenson) [inline containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:107]: Should these options live in the scss file?

**Review by emmastephenson — COMMENTED** [review]
Code changes generally look good to me, but the side nav is not rendering for me locally 🤔 I'm running through `docker-compose`, not sure if that makes a difference?

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx (hunk 1 above)
Comment (nickbristow) [reply containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:107]: I don't think they can be? https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

**Comment by nickbristow** [comment]
> @nickbristow thanks for sending the screenshot - it looks good from what I can see, but would you be able to send a video or be able to hop on a call quick (whichever is easier) to show what it looks like while you scroll? just hard to see how it's actively reflecting which section you're on in the static screenshot. thanks!

image

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx [hunk 2]
```diff
@@ -22,33 +22,134 @@ export class SectionConfig {
...
+    }
+  }
+
+  array.forEach((item) => countRecursively(item));
+
+  return count;
+}
+
+const sortHeadings = (headings: HeadingObject[]) => {
+  const result: SectionConfig[] = [];
```
Comment (KennethSkylight) [inline containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:51]: Can we add a comment about what this is doing? It's a bit confusing what it's trying to accomplish

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx [hunk 3]
```diff
@@ -22,33 +22,134 @@ export class SectionConfig {
...
+  let i = 0;
```
Comment (KennethSkylight) [inline containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:52]: Can we have a more descriptive name than "i" if possible?

**Review by KennethSkylight — COMMENTED** [review]
Just two small things

File: containers/ecr-viewer/src/app/view-data/components/tests/SideNav.test.tsx [hunk 4]
```diff
@@ -24,22 +35,21 @@ describe("SectionConfig", () => {
...
+        <SideNav />
+        <h2 id="section-1">Section 1</h2>
+        <h2 id="section-2">Section 2</h2>
+        <h3 id="section-3">Section 3</h3>
+        <h4 id="section-4">Section 4</h4>
+        <h2 id="section-2-2">Section 2 - 2</h2>
+      </>,
+    );
     expect(asFragment()).toMatchSnapshot();
   });
```
Comment (angelathe) [inline containers/ecr-viewer/src/app/view-data/components/tests/SideNav.test.tsx:50]: Thoughts on adding two unit render tests (something like, 1) if there is Immunizations data, then Immunizations show up on the SideNav & 2) if there's no Immunizations data, then Immunizations shouldn't show up on the SideNav)? or is that a bit gratuitous

Comment (nickbristow) [reply containers/ecr-viewer/src/app/view-data/components/tests/SideNav.test.tsx:53]: I added a test that specifically looks for a rendered nav link

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx (hunk 3 above)
Comment (nickbristow) [reply containers/ecr-viewer/src/app/view-data/components/SideNav.tsx]: Changed to headingIndex

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx (hunk 2 above)
Comment (nickbristow) [reply containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:52]: added a comment

**Review by KennethSkylight — APPROVED** [review]
lgtm

**Pull request merged**